src/subpackages/library/cellprofiler_library/types.py

```diff
@@ -0,0 +1,81 @@
+from multiprocessing import Value
+from tkinter import ALL
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union, Annotated, Type
+from pydantic import Field, AfterValidator, validate_call, ConfigDict
+import numpy as np
+import numpy.typing as npt
+from numbers import Number
+
+""" TODO: update this docstring
+Image Types can be classified into:
+3D/2D: "3D" or "2D"
+Single channel/multi-channel: "single" or "multi"
+Data type: uint8, uint16, float32, float64, bool
+Tiled or single: "tiled" or "single"
+"""
+
+def create_type_validator(is_3d: bool, is_multi_channel: bool, is_tiled: bool, dtype, shape_override: Optional[Iterable[int]] = None):
+    def validate(input_image: npt.NDArray[Any]):
+        print(f"running validate with args: {input_image.shape}, {is_3d}, {is_multi_channel}, {is_tiled}, {dtype}, {shape_override}")
+        # TODO: edit all error strings to be more descriptive
+        if shape_override is not None:
+            if input_image.shape != shape_override:
+                raise ValueError(f"Expected an array of shape {shape_override}, got {input_image.shape}") 
+        else:
+            if is_3d:
+                # TODO: Confirm if 3D images are inputted as 5D arrays or 3D arrays (tests seem to indicate 3D)
```

Comment:
The commented out code is for the CZTYX case. I have checked tests for 3D images for `measureimageoverlap`, `fillobjects`, and the inputs arrays are not CZTYX. Tests under `test_segmentation.py` however do follow the 5D NDArray format